src/Grpc.StatusProto/MetadataExtensions.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf;
+using Grpc.Core;
+
+namespace Grpc.StatusProto;
+
+/// <summary>
+/// Extension methods for the Grpc.Core.Metadata
+/// </summary>
+public static class MetadataExtensions  
+{
+    /// <summary>
+    /// Name of key in the metadata for the binary encoding of
+    /// <see cref="Google.Rpc.Status"/>
+    /// </summary>
+    public const string StatusDetailsTrailerName = "grpc-status-details-bin";
+
+    /// <summary>
+    /// Get the <see cref="Google.Rpc.Status"/> from the metadata.
+    /// If the metadata received contains duplicate status details then the last one found
+    /// is the one that is returned.
+    /// Note: experimental API that can change or be removed without any prior notice.
+    /// </summary>
+    /// <param name="metadata"></param>
+    /// <param name="throwOnParseError">if true then <see cref="Google.Protobuf.InvalidProtocolBufferException"/>
+    /// is thrown if the metadata cannot be parsed. Otherwise null is returned on a parsing error.</param>
+    /// <returns>
+    /// The found <see cref="Google.Rpc.Status"/> or null if it was
+    /// not present or could the data could not be parsed.
+    /// </returns>
+    public static Google.Rpc.Status? GetRpcStatus(this Metadata metadata, bool throwOnParseError=false)
```

Comment:
```suggestion
    public static Google.Rpc.Status? GetRpcStatus(this Metadata metadata, bool throwOnParseError = false)
```

Reply:
Why suppress the parse error by default? That seems like it could lead to errors disappearing.

Reply:
Copying the behaviour from Google.Api.Gax.Grpc
https://github.com/googleapis/gax-dotnet/blob/897b4bdf84d5b314c424113a8bb633bb703ff1a3/Google.Api.Gax.Grpc/RpcExceptionExtensions.cs#L73-L81

Reply:
Yes, I know this code. It feels like if we're already in an error situation (which is where this code would usually be used), failing in a different way due to parse failures in metadata is unhelpful. It's a bit like ending up with an exception caused by disposing of a socket, when the original exception is trying to *write* to the socket, if you see what I mean.

Happy to discuss alternatives though.

Reply:
I think that not losing the original error (because reporting the error itself has problems) is my preference. But no strong views.